src/com/google/enterprise/adaptor/filenet/FileNetAdaptor.java

```diff
@@ -91,11 +120,201 @@ Connection getConnection() {
         context.getSensitiveValueDecoder().decodeValue(password));
   }
 
+  private static DocId newDocId(Checkpoint checkpoint) {
+    return new DocId("pseudo/" + checkpoint);
+  }
+
+  private static DocId newDocId(Id id) {
+    return new DocId("guid/" + id);
+  }
+
   @Override
-  public void getDocIds(DocIdPusher pusher) {
+  public void getDocIds(DocIdPusher pusher) throws IOException,
+      InterruptedException {
+    pusher.pushRecords(Arrays.asList(
+        new Record.Builder(newDocId(new Checkpoint("type=document")))
```

Comment:
new Checkpoint("document", null, null) seems better. Then only Checkpoint is in the business of producing or parsing the strings.